stackrox/stackrox
# ROX-26242: Support negation in Kubernetes User Groups policy field

### Description

### Problem

**Not** checkbox is missing, even though backend supports it.

### Analysis and Solution

Add `negatedName` property so policy card renders **Not** checkbox.

### User-facing documentation


### Testing and quality


#### Automated testing


#### How I validated my change

1. `npm run lint` in ui/apps/platform
2. `npm run test` in ui/apps/platform
4. `npm run start` in ui/apps/platform

#### Manual testing

1. Visit /main/policy-management/policies/?action=create select **Runtime** and **Deployment** or **Audit logs** on **Lifecycle** step and then advance to **Rules** step.

    * Before changes without `negatedName` property, see absence of **Not** checkbox.
        ![without_negatedName](https://github.com/user-attachments/assets/376dbda0-126c-42af-93b9-bba269c87ec5)

    * After changes with `negatedName` property, see presence of **Not** checkbox.
        ![with_negatedName](https://github.com/user-attachments/assets/30a3e749-b044-4e73-b2c5-8a1175d0958e)
